src/validation.cpp

```diff
@@ -4050,39 +4050,60 @@ bool LoadMempool(void)
         if (version != MEMPOOL_DUMP_VERSION) {
             return false;
         }
-        uint64_t num;
-        file >> num;
-        double prioritydummy = 0;
-        while (num--) {
-            int64_t nTime;
-            int64_t nFeeDelta;
-            CTransaction tx(deserialize, file);
-            file >> nTime;
-            file >> nFeeDelta;
-
-            CAmount amountdelta = nFeeDelta;
-            if (amountdelta) {
-                mempool.PrioritiseTransaction(tx.GetHash(), tx.GetHash().ToString(), prioritydummy, amountdelta);
+        std::map<std::string, std::vector<unsigned char>> mapData;
+        file >> mapData;
+
+        it = mapData.find("deltas");
+        if (it == mapData.end()) {
+            try {
```

Comment:
Can't this `try` block be made to enclose all deserialization operations, so that the catch and error reporting can be written only once?